lib/improver/wind_gust_diagnostic.py

```diff
@@ -0,0 +1,227 @@
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing plugin for WindGustDiagnositic."""
+
+import warnings
+
+import iris
+from iris import FUTURE
+
+from improver.utilities.cube_manipulation import merge_cubes
+
+FUTURE.netcdf_promote = True
+
+
+class WindGustDiagnostic(object):
+
+    """Plugin for calculating wind-gust diagnostic.
+
+    In the model a shear-driven turbulence parameterization is used to
+    estimate wind gusts but in convective situations this can over-estimate the
+    convective gust.
+    This diagnostic takes the Maximum of the values at eachgrid point of
+        a chosen percentile of the wind-gust forecast and
+        a chosen percentile of the wind-speed forecast
+    to produce a better estimate of wind-gust.
+    For example a typical wind-gust could be MAX(gust(50%),windspeed(95%))
+    an extreme wind-gust forecast could be MAX(gust(95%), windspeed(100%))
+
+    See
+    https://github.com/metoppv/improver/files/1244828/WindGustChallenge_v2.pdf
+    for a discussion of the problem and proposed solutions.
+    """
+
+    def __init__(self, percentile_gust, percentile_windspeed):
+        """
+        Create a WindGustDiagnostic plugin for a given set of percentiles.
+
+        Args:
+            percentile_gust: float
+                Percentile value required from wind-gust cube.
+            percentile_windspeed: float
+                Percentile value required from wind-speed cube.
+
+        """
+        self.percentile_gust = percentile_gust
+        self.percentile_windspeed = percentile_windspeed
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<WindGustDiagnostic: wind-gust perc='
+                '{0:3.1f}, wind-speed perc={1:3.1f}>'
+                .format(self.percentile_gust, self.percentile_windspeed))
+        return desc
+
+    def add_metadata(self, cube):
+        """Add metadata to cube for windgust diagnostic.
+
+        Args:
+            cube: iris.cube.Cube instance
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result : iris.cube.Cube instance
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.standard_name = "wind_speed_of_gust"
+        result.long_name = "wind_gust_diagnostic"
+        if self.percentile_gust == 50.0 and self.percentile_windspeed == 95.0:
+            diagnostic_txt = 'Typical gusts'
+        elif (self.percentile_gust == 95.0 and
+              self.percentile_windspeed == 100.0):
+            diagnostic_txt = 'Extreme gusts'
+        else:
+            diagnostic_txt = str(self)
+        result.attributes.update({'wind_gust_diagnostic': diagnostic_txt})
+
+        return result
+
+    @staticmethod
+    def update_metadata_after_max(cube, perc_coord):
```

Comment:
Agreed, I'd scrap this and the associated unit tests. It doesn't appear to be doing anything iris can't handle.

Reply:
As I explained to Gavin earlier. The processing adds a maximum cell method which it might be appropriate to remove. I think a general decision needs to be made on cell methods but until this is decided I prefer to keep this function. Removing it will not alter the functionality of the code.

Reply:
Fine.